### PR TITLE
Reuse SupportedMediaTypes for CR content-type negotiation.

### PR DESCRIPTION
**Pull request by benluddy**
#### What this PR does / why we need it:

In the course of calling NegotiateOutputMediaType, each CR endpoint handler invocation instantiates six JSON serializers. Each instantiation marshals the serializer options to JSON to construct the serializer identifier. Under heavy CR GET load, CPU profiling shows approximately 16% of the time spent in WriteObjectNegotiated was in SupportedMediaTypes, most of that within json.identifier().

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by benluddy** [comment]
/cc @sttts @wojtek-t 

It looks like there's an existing assumption that callers must not mutate the slice returned from SupportedMediaTypes. Does this seem reasonable?

**Comment by benluddy** [comment]
cr-supportedmediatypes-flamegraph

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go [hunk 1]
```diff
@@ -833,6 +833,45 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
...
+				{
+					MediaType:        "application/vnd.kubernetes.protobuf",
+					MediaTypeType:    "application",
+					MediaTypeSubType: "vnd.kubernetes.protobuf",
+					Serializer:       protobuf.NewSerializer(creator, typer),
+					StreamSerializer: &runtime.StreamSerializerInfo{
+						Serializer: protobuf.NewRawSerializer(creator, typer),
+						Framer:     protobuf.LengthDelimitedFramer,
+					},
+				},
```
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go:873]: I know this is preexisting. But remind me why this shows up here? For the scale resource?

**Comment by sttts** [comment]
> must not mutate the slice returned from SupportedMediaTypes. Does this seem reasonable?

Yes. I would say that is even pretty common that slices returned from gettings are readonly. We should document it.

**Pull request merged**